src/shared/logger/logCleaner.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'fs-extra'
+import * as path from 'path'
+import { getLogger } from '.'
+
+export const DEFAULT_CLEANUP_PARAMS = {
+    maxLogs: 100,
+    maxFileSize: 100000000, // 100 MB
+    maxKeptLogs: 20,
+    minKeptLogs: 2,
+}
+
+/**
+ * Deletes the older logs when there are too many or are using too much space.
+ */
+export async function cleanLogFiles(logDir: string, params = DEFAULT_CLEANUP_PARAMS): Promise<void> {
```

Comment:
Just moved to its own file as it's essentially a separate sub-component.

Reply:
I suggest avoiding lots of special-purpose modules. Makes it harder to find things if there are many fine-grained files, we lose the coarse categories from which to drill down.

Reply:
I suppose a `util` would work here. But we should be keeping the `activation` file as lean as possible.